src/main/java/picard/sam/BamErrorMetric/CollectBamErrorMetrics.java

```diff
@@ -0,0 +1,470 @@
+        }
+        return regionOfInterest;
+    }
+
+    /**
+     * Compares a VariantContext to a Locus providing information regarding possible overlap, or relative location
+     *
+     * @param dictionary
+     * @param vc
+     * @param l
+     * @return negative if vc comes before l (with no overlap)
+     * zero if vc and l overlap
+     * positive if vc comes after l (with no overlap)
+     * <p/>
+     * if vc and l are in the same contig the return value will be the number of bases apart they are,
+     * otherwise it will be MIN_INT/MAX_INT
+     */
+    public static int CompareVariantContextToLocus(final SAMSequenceDictionary dictionary, final VariantContext vc, final Locus l) {
+
+        final int indexDiff = dictionary.getSequenceIndex(vc.getContig()) - l.getSequenceIndex();
+        if (indexDiff != 0) {
+            return indexDiff < 0 ? Integer.MIN_VALUE : Integer.MAX_VALUE;
+        }
+
+        //same SequenceIndex, can compare by genomic position
+        if (l.getPosition() < vc.getStart())
+            return vc.getStart() - l.getPosition();
+        if (l.getPosition() > vc.getEnd())
+            return vc.getEnd() - l.getPosition();
+        return 0;
+    }
+
+    /**
+     * Parses a "Directive" of the form "ERROR,STRATIFIER,STRATIFIER...etc." into a {@link BaseErrorAggregation} consisting of the appropriate
+     * {@link BaseErrorCalculation.BaseCalculator BaseCalculator} and the successively "paired" {@link ReadBaseStratification.RecordAndOffsetStratifier RecordAndOffsetStratifier} as listed.
+     * The conversion from string to object is performed by the enums
+     * {@link BaseErrorCalculation.Errors Errors} and {@link ReadBaseStratification.Stratifiers Stratifiers}, and the "pairing" is performed by
+     * repeatedly wrapping two stratifiers in a {@link ReadBaseStratification.PairStratifier PairStratifier}
+     *
+     * @param stratifierDirective The string directive describing the error type and collection of stratifiers to use
+     * @return The appropriate {@link BaseErrorAggregation} object.
+     */
+    protected static BaseErrorAggregation parseDirective(final String stratifierDirective) {
+        final String[] directiveUnits = new String[256];
+        final char directiveSeparator = ':';
+        final int numberOfTerms = ParsingUtils.split(stratifierDirective, directiveUnits, directiveSeparator, false);
+
+        if (numberOfTerms == 256) {
+            throw new RuntimeException("Cannot parse more than 255 terms in a single directive. Sorry. (What are you doing?)");
+        }
+        if (numberOfTerms == 0) {
+            throw new RuntimeException("Found no directives at all. Cannot process.");
+        }
+
+        // make a linkedList due to removal and addition operations below
+        final List<ReadBaseStratification.RecordAndOffsetStratifier<?>> stratifiers = new LinkedList<>(Arrays.stream(directiveUnits)
+                .limit(numberOfTerms)
```

Comment:
Why is `limit` required here if `numberOfTerms` is tested above?

Reply:
I don't know...but I get a strange error if I remove it. I'd appreciate help with this.

Reply:
I just wrote a short test without `limit` and it seemed to work OK. Are you getting a compiler error or a runtime error?

Also when I typed the code into intellij it recommended creating the linked list directly, e.g.
```
        final List<ReadBaseStratification.RecordAndOffsetStratifier<?>> stratifiers =
                Arrays.stream(directiveUnits)
                        .map(String::trim)
                        .map(ReadBaseStratification.Stratifiers::valueOf)
                        .map(ReadBaseStratification.Stratifiers::makeStratifier)
                        .collect(Collectors.toCollection(LinkedList::new));
```

Reply:
Yes...I'm getting a strange NPE if I remove that line.

Reply:
resolved. this was due to the 256-length array that was allocated.